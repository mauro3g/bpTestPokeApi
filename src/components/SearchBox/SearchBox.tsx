import React from 'react'

const style: { [key: string]: React.CSSProperties } = {
    navbar: {
        width: '100vw',
        height: '5vh',
        flexGrow: 1,
        backgroundColor: '#424242',
        color: 'white',
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'space-around',
        padding: '20px 16px'
    },
    searchBox: {
        width: '20vw',
        height: '3vh',
        fontSize: '1rem'
    },
    searchButton: {
        width: '4vh',
        height: '4vh',
        margin: '0px',

    },
    addButton: {
        color: 'white',
        textAlign: 'center',
        fontSize: '1rem',
        fontWeight: 'bold',
        padding: '7px 15px',
        cursor: 'pointer',
        border: '1px solid #fff',
        borderRadius: '5%',
        backgroundColor: '#8F8F8F',
    },
    addText: {
        margin: '0px 2px'
    }
} as const

interface Props {
    searchValue: string | undefined
    changeSearchValue: React.Dispatch<React.SetStateAction<string | undefined>>
    handleOpenForm: () => void
}

const SearchBox = (props: Props) => {
    const { searchValue, changeSearchValue, handleOpenForm } = props

    return (
        <div style={style.navbar} data-testid="search-component">
            <h4>PokeDex</h4>
            <div>
                <form onSubmit={(e) => e.preventDefault()}>
                    <input style={style.searchBox} type="text" placeholder="Buscar..." name="search" value={searchValue} onChange={(e) => changeSearchValue(e.target.value)} />
                    <button style={style.searchButton} type="submit"><i className="fa fa-search"></i></button>
                </form>
            </div>
            <button style={style.addButton} onClick={() => handleOpenForm()}>
                <i className="fa fa-plus-circle"></i>  Agregar
            </button>
        </div>
    )
}

export default SearchBox