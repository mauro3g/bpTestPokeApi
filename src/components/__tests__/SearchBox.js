import { render, screen } from '@testing-library/react';
import SearchBox from '../SearchBox/SearchBox'
import renderer from 'react-test-renderer'

test('should render a search component', () => {
    render(<SearchBox searchValue={"poke1"} changeSearchValue={()=>{}} handleOpenForm={()=>{}} />)
    const element = screen.getByTestId('search-component')
    expect(element).toBeInTheDocument()
    expect(element).toHaveTextContent("Agregar")
})
