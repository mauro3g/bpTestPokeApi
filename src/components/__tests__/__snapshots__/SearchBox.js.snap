// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshop 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#424242",
      "color": "white",
      "display": "flex",
      "flexGrow": 1,
      "height": "5vh",
      "justifyContent": "space-around",
      "padding": "20px 16px",
      "width": "100vw",
    }
  }
>
  <h4>
    PokeDex
  </h4>
  <div>
    <form
      onSubmit={[Function]}
    >
      <input
        name="search"
        onChange={[Function]}
        placeholder="Buscar..."
        style={
          Object {
            "fontSize": "1rem",
            "height": "3vh",
            "width": "20vw",
          }
        }
        type="text"
        value="poke1"
      />
      <button
        style={
          Object {
            "height": "4vh",
            "margin": "0px",
            "width": "4vh",
          }
        }
        type="submit"
      >
        <i
          className="fa fa-search"
        />
      </button>
    </form>
  </div>
  <button
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#8F8F8F",
        "border": "1px solid #fff",
        "borderRadius": "5%",
        "color": "white",
        "cursor": "pointer",
        "fontSize": "1rem",
        "fontWeight": "bold",
        "padding": "7px 15px",
        "textAlign": "center",
      }
    }
  >
    <i
      className="fa fa-plus-circle"
    />
      Agregar
  </button>
</div>
`;
